src/lwc/geBatchGiftEntryTable/geBatchGiftEntryTable.js

```diff
@@ -1,84 +1,94 @@
 import bgeActionDelete from '@salesforce/label/c.bgeActionDelete';
 import geBatchGiftsCount from '@salesforce/label/c.geBatchGiftsCount';
 import geBatchGiftsTotal from '@salesforce/label/c.geBatchGiftsTotal';
+
 import commonOpen from '@salesforce/label/c.commonOpen';
-import { isNotEmpty, isUndefined } from 'c/utilCommon';
 
-export default class GeBatchGiftEntryTable extends LightningElement {
+import DATAIMPORT_INFO from '@salesforce/schema/DataImport__c';
+import STATUS_FIELD from '@salesforce/schema/DataImport__c.Status__c';
+import FAILURE_INFORMATION_FIELD from '@salesforce/schema/DataImport__c.FailureInformation__c';
+import DONATION_AMOUNT from '@salesforce/schema/DataImport__c.Donation_Amount__c';
+
+export default class GeBatchGiftEntryTable extends GeListView {
     @api batchId;
+    @api userDefinedBatchTableColumnNames;
+    @track columnsByFieldApiName = {};
     @track ready = false;
 
     _batchLoaded = false;
     @track data = [];
-    @track hasData;
+    @track hasData = false;
 
     _columnsLoaded = false;
     _columns = [
-        {label: 'Status', fieldName: STATUS_FIELD.fieldApiName, type: 'text'},
-        {label: 'Errors', fieldName: FAILURE_INFORMATION_FIELD.fieldApiName, type: 'text'},
+        { label: 'Status', fieldName: STATUS_FIELD.fieldApiName, type: 'text' },
+        { label: 'Errors', fieldName: FAILURE_INFORMATION_FIELD.fieldApiName, type: 'text' },
         {
             label: geDonorColumnLabel, fieldName: 'donorLink', type: 'url',
-            typeAttributes: {label: {fieldName: 'donorName'}}
+            typeAttributes: { label: { fieldName: 'donorName' } }
         },
         {
             label: geDonationColumnLabel, fieldName: 'matchedRecordUrl', type: 'url',
-            typeAttributes: {label: {fieldName: 'matchedRecordLabel'}}
+            typeAttributes: { label: { fieldName: 'matchedRecordLabel' } }
         }
     ];
     _actionsColumn = {
         type: 'action',
         typeAttributes: {
             rowActions: [
-                {label: commonOpen, name: 'open'},
-                {label: bgeActionDelete, name: 'delete'}
+                { label: commonOpen, name: 'open' },
+                { label: bgeActionDelete, name: 'delete' }
             ],
             menuAlignment: 'auto'
         }
     };
-    _columnsBySourceFieldApiName = {};
-
 
     @api title;
     @api total;
     @api expectedTotal;
     @api count;
     @api expectedCount;
-    @api userDefinedBatchTableColumnNames;
     @track isLoaded = true;
 
-    @api
-    handleSectionsRetrieved(sections) {
-        if (!this._batchLoaded) {
-            this.loadBatch(sections);
-        }
+    constructor() {
+        super(DATAIMPORT_INFO.objectApiName);
+        /* Add the loadBatch function as a callback for the parent component to execute once it executes the
+        objectInfo wire service */
+        this.callbacks.push(this.loadBatch.bind(this));
```

Comment:
```suggestion
        this.callbacks.push(this.loadBatch);
```

I don't believe `.bind` is necessary here since we're working with an arrow function, the component still works without it.

Reply:
good call. done!

Reply:
It looks like this update changes the context of the call to the loadBatch method. Since it is now passed to GeListView which calls it asynchronously, it now runs anonymously - so the `this` context does need to be set explicitly to work with these changes.  

It doesn't look like we need a copy of the method, so I would try to use `.call()` rather than `.bind()`.  Even with that change, it looks like there might be some further changes needed to have this work for batch mode. For instance, `loadBatch` passes the `batchId` to the `getDataImportModel` method, but `this.batchId` won't be initialized in the constructor, so this push of the `loadBatch` method into the callbacks array in the GeListView method might need to be moved to the connectedCallback in this component.

Reply:
@kenlewis-sfdo thanks for mentioning this. i'm pretty sure we should be okay since `loadBatch` is using an arrow expression for the function and arrow expressions use the this value from the enclosing scope, which in this case is the BGE table component class. Attached a screenshot from the page load that shows the value of `this` as the BGE table and we're able to get the batchId from that even though the function executes anonymously. 

![Screen Shot 2020-06-25 at 10 48 22 AM](https://user-images.githubusercontent.com/58006245/85743109-c81d3980-b6d1-11ea-89de-205f98e856de.png)